apps/backend/src/users/users.service.ts

```diff
@@ -140,6 +142,16 @@ export class UsersService {
         'Cannot destroy only administrator account, please promote another user to administrator first'
       );
     }
+    // Clean up groups owned by user
+    await Promise.all(
```

Comment:
https://www.techiediaries.com/promise-all-map-async-await-example/

example basically doing exactly what we're doing here.  no change needed in the code, just wanted to show you